apps/builder/app/shared/nano-states/nano-states.ts

```diff
@@ -40,7 +41,32 @@ const useValue = <T>(atom: WritableAtom<T>) => {
   return [value, atom.set] as const;
 };
 
-export const selectedPageStore = atom<undefined | Page>(undefined);
+export const pagesStore = atom<undefined | Pages>(undefined);
+export const useSetPages = (pages: Pages) => {
+  useSyncInitializeOnce(() => {
+    pagesStore.set(pages);
+  });
+};
+
+export const selectedPageIdStore = atom<undefined | Page["id"]>(undefined);
+export const useSetSelectedPageId = (pageId: Page["id"]) => {
+  useSyncInitializeOnce(() => {
+    selectedPageIdStore.set(pageId);
+  });
+};
+
+export const selectedPageStore = computed(
```

Comment:
it would be better to have a  page by id map, so we can pick whatever page we need by id any time

Reply:
also think about future multi-canvas situation where you can have the different pages on different canvases visible at the same time

Reply:
> it would be better to have a page by id map, so we can pick whatever page we need by id any time

We need to move migrate away from `{ homePage, pages }` structure in db in favor of flat list. But maybe later.

> think about future multi-canvas situation

yep, state will be a little more complex